stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/attachments/factory/Permissions.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2014-2024 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.compose.ui.messages.attachments.factory
+
+import android.Manifest.permission.READ_EXTERNAL_STORAGE
+import android.Manifest.permission.READ_MEDIA_AUDIO
+import android.Manifest.permission.READ_MEDIA_IMAGES
+import android.Manifest.permission.READ_MEDIA_VIDEO
+import android.Manifest.permission.READ_MEDIA_VISUAL_USER_SELECTED
+import android.app.Activity
+import android.content.Context
+import android.os.Build
+import androidx.core.app.ActivityCompat
+
+/**
+ * Builds an [Array] of the required permissions for accessing visual media, based on the Android version.
+ */
+internal fun visualMediaPermissions(): Array<String> =
```

Comment:
Note: The permission related logic would be probably moved to the `ui-common` module so it can be used by the `XML` SDK as well.

Reply:
Yes, moving it to `ui-common` will allow us to reuse it across different artifacts and have only one piece of code where change stuff in the future